salt/states/kapacitor.py

```diff
@@ -0,0 +1,161 @@
+# -*- coding: utf-8 -*-
+'''
+Kapacitor state module.
+
+:configuration: This module accepts connection configuration details either as
+    parameters or as configuration settings in /etc/salt/minion on the relevant
+    minions::
+
+        kapacitor.host: 'localhost'
+        kapacitor.port: 9092
+
+    This data can also be passed into pillar. Options passed into opts will
+    overwrite options passed into pillar.
+
+.. versionadded:: 2016.3.0
+'''
+
+from __future__ import absolute_import
+
+import difflib
+import logging
+
+import salt.utils
+
+LOG = logging.getLogger(__name__)
+
+
+def __virtual__():
+    return 'kapacitor' if 'kapacitor.get_task' in __salt__ else  False
+
+
+def task_present(name,
+                 tick_script=None,
+                 task_type='stream',
+                 database=None,
+                 retention_policy='default',
+                 enable=True):
+    '''
+    Ensure that a task is present and up-to-date in Kapacitor.
+
+    name
+        Name of the task.
+
+    tick_script
+        Path to the TICK script for the task. Can be a salt:// source.
+
+    task_type
+        Task type. Defaults to 'stream'
+
+    database
+        Which database to fetch data from. Defaults to None, which will use the
+        default database in InfluxDB.
+
+    retention_policy
+        Which retention policy to fetch data from. Defaults to 'default'.
+
+    enable
+        Whether to enable the task or not. Defaults to True.
+    '''
+
+    comments = []
+    changes = []
+    ret = {'name': name, 'changes': {}, 'result': True, 'comment': ''}
+
+    if not tick_script:
```

Comment:
Right before I wrote that comment I was thinking along those same lines, too, but I found some examples where that isn't necessarily the case. The `file.replace` and `file.line` (which are both heavily used) both have args and kwargs.